### PR TITLE
*: support sample analyze index

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

one of split PRs for https://github.com/pingcap/tidb/pull/18717

Problem Summary:

for the non-unique local index, we cannot use the classical method to collect index's histogram after gather partition table stats, because different partitions in on table maybe contain the same data, but classical method relay on the invariant that "different index region couldn't contain the same data"

so this we need a method to collect index (one or multiple columns) value by "sample method"(that also be used to collect single column stats now)

### What is changed and how it works?

What's Changed:

- support handle `AnalyzeType_TypeSampleIndex`  in mocktikv and unistore
- support gather re-sample collect SampleIndex result from kv layer
- make unistore support analyze multiple ranges in one analyze req(supported by mocktikv and tikv)

How it Works:

### Related changes

- TiKV PR: https://github.com/tikv/tikv/pull/8443

### Check List 

Tests 

- Integration test

Side effects

- sample maybe make index stats not accurate as the classical index does

### Release note 

- No release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/19239)

### CODE REVIEW

File: statistics/sample.go [hunk 1]
```diff
@@ -145,7 +145,9 @@ func SampleCollectorFromProto(collector *tipb.SampleCollector) *SampleCollector
 	return s
 }
 
-func (c *SampleCollector) collect(sc *stmtctx.StatementContext, d types.Datum) error {
+// Collect collects sample from the result set using Reservoir Sampling algorithm,
+//// and estimates NDVs using FM Sketch during the collecting process
```
Comment (tiancaiamao) [inline statistics/sample.go:149]: ////

**Comment by tiancaiamao** [comment]
LGTM

**Comment by lysu** [comment]
@winoros @qw4990  PTAL thx

**Review by tiancaiamao — APPROVED** [review]

**Comment by lysu** [comment]
@winoros @qw4990 PTAL thx

**Comment by lysu** [comment]
@winoros @qw4990 @imtbkcat @eurekaka PTAL

**Comment by lysu** [comment]
now we choose direct reuse fast analyze sample way

**Pull request closed without merging**